shared/middleware/assets_api.rb

```diff
@@ -75,8 +77,11 @@ class AssetsApi < Sinatra::Base
 
     s3 = Aws::S3::Client.new(region: 'us-east-1')
     key = "#{CDO.assets_s3_directory}/#{owner_id}/#{channel_id}/#{filename}"
-    s3.put_object(bucket:CDO.assets_s3_bucket, key:key, body:request.body.read)
-    no_content
+    body = request.body.read
+    s3.put_object(bucket:CDO.assets_s3_bucket, key:key, body:body)
+    content_type :json
+    category = mime_type.split('/').first
```

Comment:
No validation?  We should return `415 Unsupported Media Type` for mime types we don't want to host.